src/ontology/iri_mapped.rs

```diff
@@ -199,25 +199,18 @@ impl<A: ForIRI, AA: ForIndex<A>> OntologyIndex<A, AA> for IRIMappedIndex<A, AA>
     }
 
     fn index_take(&mut self, cmp: &AnnotatedComponent<A>) -> Option<AnnotatedComponent<A>> {
-        let iris = self.aa_to_iris(cmp);
-        if !iris.is_empty() {
-            let iri = iris.iter().next();
-            if let Some(iri) = iri {
-                self.mut_set_for_iri(iri).take(cmp).map(|aax| aax.unwrap())
-            } else {
-                None
-            }
-        } else {
-            None
-        }
+        self.aa_to_iris(cmp).iter().fold(None, |val, iri| {
+            self.mut_set_for_iri(&iri.clone())
+                .take(cmp)
+                .map(|c| c.unwrap())
+                .or(val)
+        })
     }
 
     fn index_remove(&mut self, cmp: &AnnotatedComponent<A>) -> bool {
-        if let Some(iri) = self.aa_to_iris(cmp).iter().next() {
-            self.mut_set_for_iri(&iri.clone()).remove(cmp)
-        } else {
-            false
-        }
+        self.aa_to_iris(cmp).iter().fold(false, |val, iri| {
+            self.mut_set_for_iri(&iri.clone()).remove(cmp) || val
```

Comment:
`&iri.clone()` could be safely replaced with `iri` to avoid unnecessary memory allocation.